### ISSUE TITLE
[BUG] nunique with dropna=False returns incorrect results

### ISSUE THREAD

**Issue by MarcoGorelli** [rapidsai/cudf#16861]
**Describe the bug**

**Steps/Code to reproduce bug**
Follow this guide http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

```python
import cudf

df = cudf.DataFrame({"a": [1, 1, 2], "b": [4, None, 5], "c": [None, None, 7], "d": [1, 1, 3]})
df.groupby('a')['b'].nunique(dropna=False)
df.to_pandas().groupby('a')['b'].nunique(dropna=False)
```

**Expected behavior**

The ones above should match, but the cudf one is incorrect

cudf:
```
a
1    1
2    1
Name: b, dtype: int32
```

pandas:
```
a
1    2
2    1
Name: b, dtype: int64
```

Spotted using Narwhals https://github.com/narwhals-dev/narwhals 

**Environment details**
Please run and paste the output of the `cudf/print_env.sh` script here, to gather any other relevant environment details

**Additional context**

**Comment by wence-** [comment]
Thanks! This should be a simple fix, the libcudf `nunique` aggregation has an option for including or excluding nulls, but it is not plumbed through from cudf. And so it looks like the `nunique` method on groupbys just drops the `dropna` keyword argument on the floor.

**Comment by mroeschke** [comment]
Closed by https://github.com/rapidsai/cudf/pull/16921, will be fixed in the 24.12 release
